packages/flutter/test/material/text_field_test.dart

```diff
@@ -2218,6 +2218,173 @@ void main() {
     variant: TargetPlatformVariant.all(),
   );
 
+  testWidgets(
+    'Can drag the left handle while the right handle remains off-screen',
+    (WidgetTester tester) async {
+      final TextEditingController controller = TextEditingController();
+      final ScrollController scrollController = ScrollController();
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Material(
+            child: MediaQuery(
+              data: const MediaQueryData(size: Size(800.0, 600.0)),
+              child: TextField(
+                dragStartBehavior: DragStartBehavior.down,
+                controller: controller,
+                scrollController: scrollController,
+              ),
+            ),
+          ),
+        ),
+      );
+
+      // Text is longer than textfield width.
+      const String testValue =
+          'aaaaaaaaaaaaaaaaaaaaaaaaaaa bbbbbbbbbbbbbbbbbbbbbbbbbbb';
+      await tester.enterText(find.byType(TextField), testValue);
```

Comment:
In this case, you do need enterText, because testValue isn't set as the `text` in your `controller`.

Or you could potentially get rid of this and put it in the controller, like in your cupertino test above.

Reply:
I used initial text value as cupertino.